src/Features/Core/Portable/DesignerAttribute/AbstractDesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -31,12 +31,12 @@ protected AbstractDesignerAttributeIncrementalAnalyzer(Workspace workspace)
             _storageService = workspace.Services.GetRequiredService<IPersistentStorageService>();
         }
 
-        protected abstract Task ReportProjectRemovedAsync(ProjectId projectId, CancellationToken cancellationToken);
+        protected abstract ValueTask ReportProjectRemovedAsync(ProjectId projectId, CancellationToken cancellationToken);
 
-        protected abstract Task ReportDesignerAttributeDataAsync(List<DesignerAttributeData> data, CancellationToken cancellationToken);
+        protected abstract ValueTask ReportDesignerAttributeDataAsync(List<DesignerAttributeData> data, CancellationToken cancellationToken);
```

Comment:
any particular reason this changed to ValueTask?

Reply:
Yes, because the invocation delegates in RemoteConnectionAsync take ValueTask and it's easier to use VlaueTask everywhere then to convert back and forth.
Alternatively we could use Task everywhere but ValueTask will allocate less.